services/src/main/java/org/apache/druid/cli/CliOverlord.java

```diff
@@ -460,9 +460,30 @@ private void configureOverlordHelpers(Binder binder)
     );
   }
 
+
   /**
+   * Currenlty, the resource paths of the jersery resources on the overlord start with
+   *  <ol>
+   *    <li>/druid/indexer/v1</li>
+   *    <li>/druid-internal/v1</li>
+   *  </ol>
+   * <p>
+   * As QOS filtering is enabled on overlord requests, we need to update the QOS filter paths in
+   * {@link org.apache.druid.cli.CliOverlord#addQOSFiltering(ServletContextHandler, int)} when a new jersey resource is added.
    */
-  private static class OverlordJettyServerInitializer implements JettyServerInitializer
+  private static void addOverlordJerseyResources(Binder binder)
```

Comment:
Going by what other modules such as `LookupModule` are doing, we should bind the qos filter in this method itself using `JettyBindings.addQosFilter`.

Reply:
Since we donot have the exact thread pool handy, at the time of binding we can probably not do this.